AdminSelfUpgrade.php

```diff
@@ -657,8 +657,8 @@ public function init()
             $this->cleanTmpFiles();
         } else {
             foreach ($this->ajaxParams as $prop) {
-                if (property_exists($this, $prop)) {
-                    $this->{$prop} = isset($this->currentParams[$prop])?$this->currentParams[$prop]:'';
+                if (property_exists($this, $prop) && isset($this->currentParams[$prop])) {
+                    $this->{$prop} = $this->currentParams[$prop];
```

Comment:
If I understand correctly, the bug was due to the old code overwriting `$this->restoreDbFilenames` (which is supposed to be an array) with a string?

Reply:
Yes.

We have several variables in this class which are initialized on the first class, then retrieved from the params of ajax requests.
The default values are properly set on the property declaration. There is no need to apply another one if we can't find it in the request params.